services/ui/src/components/AddTask/components/DrushCron.js

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import { Mutation } from 'react-apollo';
+import gql from 'graphql-tag';
+import ReactSelect from 'react-select';
+import { bp, color, fontSize } from 'lib/variables';
+
+const taskDrushCron = gql`
+  mutation taskDrushCron(
+    $environment: Int!
+  ) {
+    taskDrushCron(
+      environment: $environment
+    ) {
+      id
+      name
+      status
+      created
+      started
+      completed
+      remoteId
+      command
+      service
+    }
+  }
+`;
+
+const DrushCron = ({
+  pageEnvironment,
+  onCompleted,
+  onError,
+}) => (
+  <Mutation
+    mutation={taskDrushCron}
+    onCompleted={onCompleted}
+    onError={onError}
+  >
+    {(taskDrushCron, { loading, called, error, data }) => {
+      return (
+        <React.Fragment>
+          <div className="envSelect">
+            <label id="dest-env">Environment:</label>
+            <ReactSelect
+              aria-labelledby="dest-env"
+              name="dest-environment"
+              value={{
+                label: pageEnvironment.name,
+                value: pageEnvironment.id,
+              }}
+              options={[
+                {
+                  label: pageEnvironment.name,
+                  value: pageEnvironment.id,
+                }
+              ]}
+              isDisabled
+              required
+            />
+          </div>
+          <button
+            onClick={() =>
+              taskDrushCron({
+                variables: {
+                  environment: pageEnvironment.id
+                }
+              })
+            }
```

Comment:
Would be good to have protection in the UI against hitting the button repeatedly

Reply:
Got a change ready for this one, I just need to test it first.

Reply:
What's a sensible protection here.  I was originally thinking placing an elapsed time check in the task itself, but realise that setting it to more than a minute may negate the viability of the task.

Reply:
It should be noted that the other tasks don't have a mechanism to delay based on usage.

I think it's a great idea but I believe that should not be controlled by the individual task, but where the is actually evaluated. The vector already exists - so shouldn't the solution for that problem be made generic enough and implementable for all tasks...?

Reply:
Sounds good, a new issue sounds like a great. The main idea from my perspective is to prevent accidental double clicks.

Reply:
I'll get this change tested tonight, and submit a new issue for adding task constraints.

Reply:
**Step 1**
![image](https://user-images.githubusercontent.com/3070264/67711982-31e82500-fa17-11e9-96e7-3220c72dd019.png)

**Step 2**
![image](https://user-images.githubusercontent.com/3070264/67712029-47f5e580-fa17-11e9-933f-ee7ac755530f.png)

**Step 3**
![image](https://user-images.githubusercontent.com/3070264/67712048-547a3e00-fa17-11e9-9726-5542cca22fc4.png)

Reply:
@seanhamlin, would you consider this review as resolved, or is there more work to do?